core/lib/config/src/configs/observability.rs

```diff
@@ -0,0 +1,12 @@
+/// Configuration for the essecial observability stack, like
+/// logging and sentry integration.
+#[derive(Debug, Clone)]
+pub struct ObservabilityConfig {
+    /// URL of the Sentry instance to send events to.
+    pub sentry_url: Option<String>,
+    /// Name of the environment to use in Sentry.
+    pub sentry_environment: Option<String>,
+    /// Format of the logs as expected by the `vlog` crate.
+    /// Currently must be either `plain` or `json`.
+    pub log_format: String,
```

Comment:
I could've made it an enum, but it felt a bit off; having `impl FromStr for vlog::LogFormat` felt enough for that purpose.

Reply:
You do it every time anyway, so why not just do it here?

Reply:
What do you mean? Make `zksync_config` depend on `vlog`? This will make any crate that depends on `zksync_config` (most of which are libraries) transitively depend on sentry and tracing-subscriber. Make `vlog` depend on `zksync_config`? 
This also doesn't feel right, since we're trying to move away from specific crates being aware of the whole stack.

What we can do is to introduce a `enum LogFormat { Plain, Json }` here and do the conversion of `zksync_config::LogFormat` to `vlog::LogFormat`, but we'll still have to do it each time.

Another option would be to introduce one more "common" crate for `LogFormat` enum, but that'd _certainly_ be an overkill.

Reply:
Invalid in what sense? You can provide a wrong sentry URL as well. Your code will just fail not at the phase of config deserialization, but when you will try to parse the log format - which is generally the very next thing you would do after deserializing the config. And similarly, that'd also be the step where the code will fail if you'll provide the invalid Sentry URL, so it's pretty similar in that sense.